shell/src/main/java/alluxio/cli/bundler/command/CollectLogCommand.java

```diff
@@ -71,4 +308,34 @@ public String getUsage() {
   public String getDescription() {
     return "Collect Alluxio log files";
   }
+
+  /**
+   * Identifies the datetime from a certain piece of log by trying various known patterns.
+   * Returns null if unable to identify a datetime.
+   *
+   * @param s a log entry
+   * @return identified datetime
+   * */
+  @Nullable
+  public static LocalDateTime parseDateTime(String s) {
+    for (Map.Entry<String, Integer> entry : FORMAT_TO_LEN.entrySet()) {
+      String f = entry.getKey();
+      int len = entry.getValue();
```

Comment:
Is `String.length()` slow?

Reply:
The thing is, the real length of the datetime string like "2020-07-25T12:05:12.123+0800" is not the same as length of the pattern "yyyy-MM-dd'T'HH:mm:ss.SSSXX". Note that the single quotes around `T` should not be counted and `XX` expands to "+0800".

Yea I should add examples to these formats.